src/shogun2-util/src/main/java/de/terrestris/shogun2/util/interceptor/MutableHttpServletRequest.java

```diff
@@ -0,0 +1,184 @@
+package de.terrestris.shogun2.util.interceptor;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.ServletInputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * An implementation of HttpServletRequestWrapper.
+ *
+ * @author Daniel Koch
+ *
+ * See: http://stackoverflow.com/questions/10210645/http-servlet-request-lose-params-from-post-body-after-read-it-once
+ */
+public class MutableHttpServletRequest extends HttpServletRequestWrapper {
+
+	/**
+	 * Holds custom parameter mapping
+	 */
+	private Map<String, String[]> customParameters;
+
+	/**
+	 *
+	 */
+	private String customRequestURI;
+
+	/**
+	 *
+	 */
+	private ByteArrayOutputStream cachedInputStream;
```

Comment:
Hmmm. Why do we have a ByteArray**Output**Stream in property cached**Input**Stream. Do I miss sometheing?